src/app/checkout/getCheckoutStepStatuses.ts

```diff
@@ -77,14 +77,26 @@ const getShippingStepStatus = createSelector(
 
         return shippingAddress ? data.getShippingAddressFields(shippingAddress.countryCode) : EMPTY_ARRAY;
     },
-    (shippingAddress, consignments, cart, payment, shippingAddressFields) => {
+    ({ data }: CheckoutSelectors) => data.getConfig(),
+    (shippingAddress, consignments, cart, payment, shippingAddressFields, config) => {
         const hasAddress = shippingAddress ? isValidAddress(shippingAddress, shippingAddressFields) : false;
         // @todo: interim solution, ideally we should render custom form fields below amazon shipping widget
         const hasRemoteAddress = !!shippingAddress && !!payment && payment.id === 'amazon';
         const hasOptions = consignments ? hasSelectedShippingOptions(consignments) : false;
         const hasUnassignedItems = cart && consignments ? hasUnassignedLineItems(consignments, cart.lineItems) : true;
         const isComplete = (hasAddress || hasRemoteAddress) && hasOptions && !hasUnassignedItems;
-        const isRequired = cart ? cart.lineItems.physicalItems.some(lineItem => lineItem.isShippingRequired) : false;
+        const isRequired = (() => {
+            if (cart) {
+                const physicalItems = cart.lineItems.physicalItems.some(lineItem => lineItem.isShippingRequired);
+                if (config && config.checkoutSettings.features['CHECKOUT-4936.enable_custom_item_shipping'] && cart.lineItems.customItems) {
+                    return cart.lineItems.customItems.length > 0 || physicalItems;
+                }
+
+                return physicalItems;
+            }
+
+            return false;
+        })();
```

Comment:
rather than creating a function that calls itself, maybe create an actual function in a new file? Or even re-use the function `getShippableItemsCount` and modify so that it can take a second param

`getShippableItemsCount(cart, includeCustomItems: bool = false)`

Reply:
I opted to write a new function in a separate file, since the intent is to check if anything in cart requires shipping, versus counting the shippable items.